test/python/circuit/test_control_flow.py

```diff
@@ -0,0 +1,383 @@
+        """Verify creation and properties of a BreakLoopOp."""
+        op = BreakLoopOp(3, 1)
+
+        self.assertIsInstance(op, Instruction)
+        self.assertEqual(op.name, "break_loop")
+        self.assertEqual(op.num_qubits, 3)
+        self.assertEqual(op.num_clbits, 1)
+        self.assertEqual(op.params, [])
+
+
+@ddt
+class TestAddingControlFlowOperations(QiskitTestCase):
+    """Tests of instruction subclasses for dynamic QuantumCircuits."""
+
+    @data(
+        (Clbit(), True),
+        (ClassicalRegister(3, "test_creg"), 3),
+        (ClassicalRegister(3, "test_creg"), True),
+    )
+    def test_appending_while_loop_op(self, condition):
+        """Verify we can append a WhileLoopOp to a QuantumCircuit."""
+        body = QuantumCircuit(3, 1)
+        body.add_register([condition[0]] if isinstance(condition[0], Clbit) else condition[0])
+
+        op = WhileLoopOp(condition, body)
+
+        qc = QuantumCircuit(5, 2)
+        qc.append(op, [1, 2, 3], [1])
+
+        self.assertEqual(qc.data[0][0].name, "while_loop")
+        self.assertEqual(qc.data[0][0].params, [body])
+        self.assertEqual(qc.data[0][0].condition, condition)
+        self.assertEqual(qc.data[0][1], qc.qubits[1:4])
+        self.assertEqual(qc.data[0][2], [qc.clbits[1]])
+
+    @data(
+        (Clbit(), True),
+        (ClassicalRegister(3, "test_creg"), 3),
+        (ClassicalRegister(3, "test_creg"), True),
+    )
+    def test_quantumcircuit_while_loop(self, condition):
+        """Verify we can append a WhileLoopOp to a QuantumCircuit via qc.while_loop."""
+        body = QuantumCircuit(3, 1)
+        body.add_register([condition[0]] if isinstance(condition[0], Clbit) else condition[0])
+
+        qc = QuantumCircuit(5, 2)
+        qc.while_loop(condition, body, [1, 2, 3], [1])
+
+        self.assertEqual(qc.data[0][0].name, "while_loop")
+        self.assertEqual(qc.data[0][0].params, [body])
+        self.assertEqual(qc.data[0][0].condition, condition)
+        self.assertEqual(qc.data[0][1], qc.qubits[1:4])
+        self.assertEqual(qc.data[0][2], [qc.clbits[1]])
+
+    def test_appending_for_loop_op(self):
+        """Verify we can append a ForLoopOp to a QuantumCircuit."""
+        body = QuantumCircuit(3, 1)
+        loop_parameter = Parameter("foo")
+        indexset = range(0, 10, 2)
+
+        body.rx(loop_parameter, [0, 1, 2])
+
+        op = ForLoopOp(loop_parameter, indexset, body)
+
+        qc = QuantumCircuit(5, 2)
+        qc.append(op, [1, 2, 3], [1])
+
+        self.assertEqual(qc.data[0][0].name, "for_loop")
+        self.assertEqual(qc.data[0][0].params, [loop_parameter, indexset, body])
+        self.assertEqual(qc.data[0][1], qc.qubits[1:4])
+        self.assertEqual(qc.data[0][2], [qc.clbits[1]])
+
+    def test_quantumcircuit_for_loop_op(self):
+        """Verify we can append a ForLoopOp to a QuantumCircuit via qc.for_loop."""
+        body = QuantumCircuit(3, 1)
+        loop_parameter = Parameter("foo")
```

Comment:
It'd be good to test the failure paths for parameter setting as well, if not only so in the future we've got an illustration of what we intended to be an error/warning.

Reply:
Agree, added in f681c37c .